### ISSUE TITLE
Install Problems

### ISSUE THREAD

**Issue by EthyLARSindichlorid** [adepierre/SniffCraft#14]
I never worked with cmake, Visual Studio or C++. Can someone please help me install the programm? I tried it severel times and after finishing all steps and trying to run the sniffcraft command in cmd, i get the message that the command does not exist.

**Comment by adepierre** [comment]
Hello!

Could you detail the steps you did? If you've compiled the program successfully, you should have a sniffcraft.exe program in a ``bin`` folder (either by double clicking it or by launching it through a cmd prompt with the absolute path specified).

To open a cmd prompt at the same location than the exe, you can use the windows explorer: go to the folder you want to be in (the one containing sniffcraft.exe), click on the address bar on the top (like in a web browser), type in cmd and press enter. It should open a cmd prompt already at this location.

**Comment by EthyLARSindichlorid** [comment]
Thanks, found the .exe. But i have a new Problem. When i start the .exe, i get an error Message, that zlib.dll could not be found. My instalation steps:
1. cloned the repository with cmd in a Folder
2. used cmake-gui to build the project
3. used Visual Studio 2022 to compile the project

Everything is done in my user account. Do i need adminrights in one step to install it correctly?

**Comment by adepierre** [comment]
Ok I know what the issue is, someone had the same on my other botcraft repo. Please check [this answer](https://github.com/adepierre/Botcraft/issues/88#issuecomment-1166526474) to see if you can copy the dll. I'll try to integrate the fix I made for botcraft into sniffcraft later.

**Comment by EthyLARSindichlorid** [comment]
Thanks, works now. I wanted to use the PacketLogger for a modded 1.18.2 server. But when i try to connect to the server i get the message that the server has forge installed and the client needs mods installed. If i try to connect without the proxy i works. Is there a solution to use the proxy with modded minecraft?

**Comment by adepierre** [comment]
It first targets vanilla server so I'm not really surprised it doesn't work as is with a modded one. Could you share the config you are using so I can try to reproduce the issue (server jar link/mods to install). Thanks

**Comment by EthyLARSindichlorid** [comment]
I use All the Mods 7 version 0.4.21 on server and client, with some additional mods. (The server uses an address like: my.server.de:port). The additional mods are: 
alloyed-1.18.2-v1.4b.jar
connectivity-1.18.2-3.2.jar
createaddition-1.18.2-20220715a.jar
createdeco-1.2.8-1.18.2.jar
create-stuff-additions1.18.2_v1.1.8.jar
Cyclic-1.18.2-1.7.2.jar

If you need additional information to reproduce the issue let me know.

**Comment by EthyLARSindichlorid** [comment]
Here you can download the server and mods for the server. There schould be a readme file if you need help how to setup forge and the server: https://www.curseforge.com/minecraft/modpacks/all-the-mods-7/files/3873165

The client is a bit more complicated. There are multiple ways. The easiest solution would be to use the Cursforge app from Overwolf. Another solution would be to use MultiMC. I didnt setup a modded client on the normal Minecraft launcher for a long time and dont really know how to do it there.

If you want i could send you a Google-Drive link to download a setuped server and a MultiMc profile so you can just import the profile in MultiMc and start the modpack. If you have Discord i can help you setup the modded server and client, when i am at home.

Here you can download MultiMc: https://multimc.org/#Download
And here the Curseforge-App: https://download.curseforge.com/

If you want a detailed step by step instruction how to setup your client and the server, i can make you one, when i am at home.

**Comment by adepierre** [comment]
Should be fine with the links, thanks. I'll try it and contact you if I don't manage to reproduce the issue.

**Comment by Neboer** [comment]
I suggest you should provide prebuild binaries to github release.

**Comment by adepierre** [comment]
> I suggest you should provide prebuild binaries to github release. The install process is very difficult and I have encountered several problems which I don't know how to deal with.

I've made a detailed tutorial for botcraft (see [here](https://github.com/adepierre/Botcraft/wiki/Detailed-installation-guide)). It's not exactly the same, but could be helpful.

Would you mind open another issue with the problems you ran into? Maybe I could help you figure it out? I'd like to keep this one about the Forge stuff that I still have to investigate.

About the release, I think that would be possible for sniffcraft. The only issue I see is that I'd need to provide 24 different binaries for each versions of the game so it might be a bit messy.

**Comment by adepierre** [comment]
@EthyLARSindichlorid Sorry for the delay, I've been quite busy with the 1.19.1/1.19.2 updates.

Anyway, I finally got time to setup the server and client to reproduce your bug. I got both of them running correctly (at least I think). However, they take quite a lot of ressources (CPU/RAM), and whenever I try to join the server with the client running on the same computer, I just got a "not responding" window and nothing happens (even after a few minutes).

I'm afraid there is not much I can do as I can't even connect to the server with the vanilla client. Is there something else I should know to make it work?
